libraries/AP_GSOF/AP_GSOF.h

```diff
@@ -0,0 +1,138 @@
+// Trimble GSOF protocol parser library.
+
+#pragma once
+
+#include "AP_GSOF_config.h"
+
+#if AP_GSOF_ENABLED
+
+class AP_GSOF
+{
+public:
+
+    static constexpr int NO_GSOF_DATA = 0;
+    static constexpr int UNEXPECTED_NUM_GSOF_PACKETS = -1;
+
+    // Parse a single byte into the buffer.
+    // When enough data has arrived, it returns the number of GSOF packets parsed in the GENOUT packet.
+    // If an unexpected number of GSOF packets were parsed, returns UNEXPECTED_NUM_GSOF_PACKETS.
+    // This means there is no fix.
+    // If it returns NO_GSOF_DATA, the parser just needs more data.
+    int parse(const uint8_t temp, const uint8_t n_expected) WARN_IF_UNUSED;
+
+    // GSOF packet numbers.
+    enum msg_idx_t {
+        POS_TIME = 1,
+        POS = 2,
+        VEL = 8,
+        DOP = 9,
+        POS_SIGMA = 12
+    };
+
+    // GSOF1
+    struct PACKED pos_time_t {
+        uint32_t time_week_ms;
+        uint16_t time_week;
+        uint16_t num_sats;
+        uint8_t pos_flags1;
+        uint8_t pos_flags2;
+        uint8_t initialized_number;
+    };
+
+    pos_time_t pos_time;
+
+    // GSOF2
+    struct PACKED pos_t {
+        // [rad]
+        double latitude;
```

Comment:
latitude expressed in radians! I'd definitely put a _rad on the end of the variable name.

Reply:
Done